nft-simple/src/nft_core.rs

```diff
@@ -133,23 +144,45 @@ impl NonFungibleTokenCore for Contract {
     }
 
     #[payable]
-    fn nft_approve_account_id(&mut self, token_id: TokenId, account_id: ValidAccountId) -> bool {
-        assert_one_yocto();
-        let mut token = self.tokens_by_id.get(&token_id).expect("Token not found");
-        assert_eq!(&env::predecessor_account_id(), &token.owner_id);
+    fn nft_approve(
+        &mut self,
+        token_id: TokenId,
+        account_id: ValidAccountId,
+        msg: Option<String>,
+    ) {
         let account_id: AccountId = account_id.into();
         let storage_used = bytes_for_approved_account_id(&account_id);
-        if token.approved_account_ids.insert(account_id) {
-            deposit_refund(storage_used);
+        assert!(env::attached_deposit() >= storage_used as u128, "attached_deposit doesn't cover storage of account_id: {}", account_id.clone());
+
+        let mut token = self.tokens_by_id.get(&token_id).expect("Token not found");
+        assert_eq!(&env::predecessor_account_id(), &token.owner_id);
+
+        if token.approved_account_ids.insert(account_id.clone()) {
             self.tokens_by_id.insert(&token_id, &token);
-            true
-        } else {
-            false
+            deposit_refund(storage_used);
+        }
+
```

Comment:
Make sure you refund everything in case it was already approved

Reply:
see https://github.com/near/core-contracts/pull/135/commits/951e6ef9feb6578b47228c8f9848c53aaa75dbfb